pkg/orchestrator/assetscanwatcher/watcher.go

```diff
@@ -195,8 +195,8 @@ func (w *Watcher) reconcilePending(ctx context.Context, assetScan *models.AssetS
 	if assetScan.Scan != nil && assetScan.Scan.Id != "" && assetScan.Scan.MaxParallelScanners != nil {
 		// Check whether we have reached the maximum number of running scans
 		// TODO(chrisgacsal): the number of concurrent scans needs to be part of the provider config and handled there
-		filter := fmt.Sprintf("scan/id eq '%s' and status/general/state ne '%s' and status/general/state ne '%s' and resourceCleanup eq '%s'",
-			assetScan.Scan.Id, models.AssetScanStateStateDone, models.AssetScanStateStatePending, models.ResourceCleanupStatePending)
+		filter := fmt.Sprintf("scan/id eq '%s' and status/general/state ne '%s' and status/general/state ne '%s' and resourceCleanup/state eq '%s'",
+			assetScan.Scan.Id, models.AssetScanStateStateDone, models.AssetScanStateStatePending, models.ResourceCleanupStatusStatePending)
```

Comment:
Can you please explain why only resources with cleanup state pending are considered to be running?

Reply:
The reason for this is we want to make sure that resource cleanup also taken into consideration when determining the number of allowed assets being scanned to prevent exceeding the resources certain providers have.

This is a result of a long discussion with @Tehsmash when resource cleanup was introduced.